File: Tours/src/main/java/com/soa/toursservice/model/OrderItem.java
package com.soa.toursservice.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long tourId;

    private String tourName;

    private double price;

    @ManyToOne
    @JoinColumn(name = "shopping_cart_id", nullable = false)
    @JsonIgnore
    private ShoppingCart shoppingCart;

    public OrderItem() {
    }

    public Long getId() {
        return id;
    }

    public Long getTourId() {
        return tourId;
    }

    public void setTourId(Long tourId) {
        this.tourId = tourId;
    }

    public String getTourName() {
        return tourName;
    }

    public void setTourName(String tourName) {
        this.tourName = tourName;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public ShoppingCart getShoppingCart() {
        return shoppingCart;
    }

    public void setShoppingCart(ShoppingCart shoppingCart) {
        this.shoppingCart = shoppingCart;
    }
}

File: Purchase/Purchase.Domain/Entities/OrderItem.cs
using System.Text.Json.Serialization;

namespace Purchase.Domain.Entities;

public class OrderItem
{
    public long Id { get; set; }

    public long TourId { get; set; }

    public string TourName { get; set; } = string.Empty;

    public double Price { get; set; }

    public long ShoppingCartId { get; set; }

    [JsonIgnore]
    public ShoppingCart ShoppingCart { get; set; } = null!;
}